### PR TITLE
[gpa] use value receivers for Hash

### PR DESCRIPTION
**Pull request by dessaya**

### CODE REVIEW

File: packages/state/l1commitment.go [hunk 1]
```diff
@@ -132,7 +132,7 @@ func (s *L1Commitment) Read(r io.Reader) error {
 }
 
 func (s *L1Commitment) String() string {
-	return fmt.Sprintf("trie root: %s, block hash: %s", s.trieRoot.String(), s.GetBlockHash())
+	return fmt.Sprintf("trie root: %s, block hash: %s", s.GetTrieRoot(), s.GetBlockHash())
```
Comment (kape1395) [inline packages/state/l1commitment.go:135]: Is this change intended?

Comment (dessaya) [reply packages/state/l1commitment.go:135]: Yes :) Go now automatically calls String()

Comment (kape1395) [reply packages/state/l1commitment.go:135]: It was not calling it, probably because of the pointer/value receiver mismatch.
